### PR TITLE
IOOP-39 Enable client registration endpoint

### PR DESCRIPTION
**Pull request by datalek**

### CODE REVIEW

**Review by kin0992 — CHANGES_REQUESTED** [review]
Just to use the same pattern used for the client_id

File: README.md [hunk 1]
```diff
@@ -48,15 +48,35 @@ Those are all Environment variables needed by the application:
...
+3. Open the browser on the following endpoint `http://localhost:3000`, and add the following cookie:
+
+```
+X-IO-Federation-Token=<any-value>
+```
+
+4. In the same browser session paste the following endpoint replacing the `<client_id>` with copied `client_id` value and then :
+
+```
+http://localhost:3000/oauth/authorize?client_id=<client_id>&response_type=id_token&redirect_uri=https%3A%2F%2Fclient.example.org%2Fcallback&scope=openid&state=the-state-here&nonce=the-nonce-here
```
[the remark this answers is not in the document]
Comment (datalek) [reply README.md]: Done 👍

**Review by kin0992 — APPROVED** [review]
Nice! 🍻

File: README.md [hunk 2]
```diff
@@ -48,15 +48,35 @@ Those are all Environment variables needed by the application:
...
+``` sh
+curl --location --request POST 'http://localhost:3000/connect/register' \
+--header 'Content-Type: application/json' \
+--data-raw '{
+   "redirect_uris":
+     ["https://client.example.org/callback"],
+   "client_name": "This is a test client",
+   "grant_types": ["implicit"],
+   "response_types": ["id_token"],
+   "token_endpoint_auth_method": "none"
```
Comment (francescopersico) [inline README.md:67]: Can we specify the scopes?

Comment (datalek) [reply README.md:67]: Yep! With `scope` attribute. And should match the values provided in [provider.ts#L94](https://github.com/pagopa/io-openid-provider/blob/2b8ece52f981ac39270fd23cffe57cb9346163c8/src/oidcprovider/provider.ts#L94).

Example: 

``` sh
curl --location --request POST 'http://localhost:3000/connect/register' \
--header 'Content-Type: application/json' \
--data-raw '{
   "redirect_uris":
     ["https://client.example.org/callback"],
   "client_name": "This is a test client",
   "grant_types": ["implicit"],
   "response_types": ["id_token"],
   "token_endpoint_auth_method": "none",
   "scope": "openid"
}'
```

## Refs
1. https://datatracker.ietf.org/doc/html/rfc7591#section-2

**Review by francescopersico — APPROVED** [review]

**Pull request merged**